src/D2L.Bmx/OktaAuthenticator.cs

```diff
@@ -130,6 +140,103 @@ private bool TryAuthenticateFromCache(
 		return true;
 	}
 
+	private async Task<IOktaAuthenticatedClient?> TryAuthenticateWithDSSOAsync(
+		string org,
+		string user,
+		IOktaClientFactory oktaClientFactory,
+		bool experimentalBypassBrowserSecurity
+	) {
+		await using IBrowser? browser = await Browser.LaunchBrowserAsync( experimentalBypassBrowserSecurity );
+		if( browser is null ) {
+			return null;
+		}
+
+		Console.WriteLine( "Attempting to automatically login using DSSO." );
+		var cancellationTokenSource = new CancellationTokenSource( TimeSpan.FromSeconds( 15 ) );
+		var sessionIdTaskProducer = new TaskCompletionSource<string?>( TaskCreationOptions.RunContinuationsAsynchronously );
+		string? sessionId;
+
+		try {
+			var page = await browser.NewPageAsync().WaitAsync( cancellationTokenSource.Token );
+			string baseAddress = org.Contains( '.' )
+				? $"https://{org}/"
+				: $"https://{org}.okta.com/";
+			var baseUrl = new Uri( baseAddress );
```

Comment:
https://github.com/Brightspace/bmx/pull/480#discussion_r1764035285
I was thinking we'd construct the Okta base URL once in OktaAuthenticator and pass it to the oktaClientFactory and the browser, instead of (re)constructing it from the user-supplied "org" value every time.